students/jstonham/README.md

```diff
@@ -2752,4 +2810,68 @@ Cookies
 
 Bandwidth
-- Bandwidth measures volume of data in bits per second.
\ No newline at end of file
+- Bandwidth measures volume of data in bits per second.
+
+## OSI Model
+
+The International Standards Organisation (ISO) developed the Open Systems Interconnection (OSI) model.
+The OSI model is framework that defines how communication systems should operate regardless of their underlying structure.
+The model divides data communication into seven layers that group protocols by their functionality, which ensures that all systems communicate data in a similar way.
+
+The seven layers of the [OSI Model](https://www.webopedia.com/quick_ref/OSI_Layers.asp) are:
+
+1. Physical
+1. Data Link
+1. Network
+1. Transport
+1. Session
+1. Presentation
+1. Application
+
+However, the layers are normally shown in the reverse order as sending data starts with an application and moves down through the layers, with no-longer-needed data being stripped away until the physical transport layer.
+From this, layers are then added back on until the application layer, at which point the data has made it to the receiver.
+
+| OSI Layer | Examples |
+| --------- | -------- |
+| 1. Physical | Ethernet |
+| 2. Data Link | PPP |
+| 3. Network | IP |
+| 4. Transport | TCP |
+| 5. Session | SQL |
+| 6. Presentation | GIF, JPEG |
+| 7. Application | WWW browsers, HTTP |
+
+The OSI Model is the basic standard for data communication, but some layers can be grouped together and data can be sent and received without needing all of the layers.
+
+## More on Protocols
+
+- The Internet Protocol tells two computers how to find each other.
+- Transmission Control Protocol (TCP) allows the computers to have a connection.
+- HTTP defines the `request-response` paradigm.
+- Git is a protocol that can be used on top of HTTP.
+- Postman is an HTTP client.
+- A web server is something that responds to requests.
+- A mail server responds to email requests.
+
+## Compilers and Transpilers
+
+Compilers are programs that take source code written in one language and produce an executable output file in another language.
+
+Transpilers take a source code file and convert it to a source code file in another language or a different version of the same language.
+The output will not be executable until it goes through a compiler.
+Transpilers are also called source-to-source compilers.
+
+## Model-View-Controller (MVC) Framework
+
+The MVC framework is an architectural design used for developing user interfaces where the application is divided into three interconnected parts.
+
+Model - Data stored in a database.
+
+View - Data shown to the user.
+
+Controller - Business logic.
+
+In MVC frameworks, the model and view are decoupled.
+This means they don't interact with each other and can therefore be updated independently of each other.
+
+Examples of MVC frameworks are Rails (written in Ruby) and Sails.js (written in Node.js).
```

Comment:
I took the liberty of using a more known framework (I've never even heard of sails).

```suggestion
Examples of MVC frameworks are Rails (written in Ruby) and Spring MVC (written in Java).
```